dockstore-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java

```diff
@@ -618,6 +618,47 @@ public void testImageIDUpdateDuringRefresh() throws ApiException {
         assertTrue(size2 == 0 && size >= 1);
     }
 
+    /**
+     * Tests that image and checksum information can be grabbed from Quay and update db correctly.
+     */
+    @Test
+    public void testGrabbingImagesFromQuay() {
+        ContainersApi containersApi = setupWebService();
+        DockstoreTool tool = getContainer();
+        tool.setRegistry(DockstoreTool.RegistryEnum.QUAY_IO);
+        tool.setNamespace("dockstoretestuser2");
+        tool.setName("dockstore-tool-imports");
+        tool.setMode(DockstoreTool.ModeEnum.AUTO_DETECT_QUAY_TAGS_AUTOMATED_BUILDS);
+        tool = containersApi.registerManual(tool);
+
+        assertEquals(0, containersApi.getContainer(tool.getId(), null).getWorkflowVersions().get(0).getImages().size());
+
+        UsersApi usersApi = new UsersApi(containersApi.getApiClient());
+        final Long userid = usersApi.getUser().getId();
+        usersApi.refresh(userid);
+
+        // Check that the image information has been grabbed on refresh.
+        List<Tag> tags = containersApi.getContainer(tool.getId(), null).getWorkflowVersions();
+        for (Tag tag : tags) {
+            assertNotNull(tag.getImages().get(0).getChecksums().get(0).getType());
```

Comment:
Too late, but can a not null string simply be an empty string (which would not be what we expect)